GitUI/UserControls/NativeTreeView.cs

```diff
@@ -8,4 +12,100 @@ protected override void CreateHandle()
             NativeMethods.SetWindowTheme(Handle, "explorer", null);
         }
     }
+
+    // Add double-click related behaviour to NativeTreeView
+    public class NativeTreeViewDoubleClickDecorator
```

Comment:
I'd like to have unit tests for this class. It looks very testable with exception of `DateTime.Now` construct.
To accommodate this, we can add either another ctor with the following signature:
```
public NativeTreeViewDoubleClickDecorator(NativeTreeView treeView, Func<DateTime> getDateTime)
{
}
public NativeTreeViewDoubleClickDecorator(NativeTreeView treeView)
    : this(treeView, () => DateTime.Now)
{
}
```
or use a `TestAccessor` pattern to set a local field `Func<DateTime> _getDateTime`.

Reply:
I'll go with the former.

Reply:
`getCurrentTime`?